### PR TITLE
styles: add new mobile menu icon

### PR DESCRIPTION
**Pull request by leandrogavidia**
## Summary by Sourcery

Add a new `MobileNavIcon` component to replace the existing animation library for the mobile menu icon, enhancing the customization and control over the icon's appearance and behavior.

New Features:
- Introduce a new `MobileNavIcon` component to handle the mobile menu icon display and interaction.

Enhancements:
- Replace the `UseAnimations` component with the new `MobileNavIcon` component in the mobile navigation for a more customized icon animation.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `MobileNavIcon` component for improved mobile navigation, visually indicating the open or closed state of the menu.
	- Added new navigation links: "My Wallet" and "Sign out" in the `NavLinks` component.
	- Updated the "Pre Sale" link text to "Presale" and adjusted its disabled condition.
- **Refactor**
	- Updated the `MobileNav` component to utilize the new `MobileNavIcon`, streamlining the navigation icon rendering.
- **Chores**
	- Updated the `framer-motion` package to version `^11.5.5`.
	- Removed the `react-useanimations` package from dependencies.
	- Changed the default formatter for TypeScript React files in the editor settings.

### CODE REVIEW

**Comment by AndlerRL** [comment]
@leandrogavidia -- can you pull the latest changes from develop? The TS issue has been fixed already.

File: .vscode/settings.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "editor.defaultFormatter": "biomejs.biome"
   },
   "[typescriptreact]": {
-    "editor.defaultFormatter": "biomejs.biome"
+    "editor.defaultFormatter": "vscode.typescript-language-features"
```
[the remark this answers is not in the document]
Comment (leandrogavidia) [reply .vscode/settings.json:14]: @AndlerRL Done :D

**Comment by leandrogavidia** [comment]
@AndlerRL Done

File: .vscode/settings.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndlerRL) [reply .vscode/settings.json:14]: @leandrogavidia -- can we revert this to use biomejs config? 🙏 Only this and it should be good to go!

[the remark this answers is not in the document]
Comment (leandrogavidia) [reply .vscode/settings.json]: Done, sorry for that

File: apps/web/components/layout/header/mobile-nav-icon/index.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+"use client"
+
+import { cn } from "@/lib/utils"
+
+export function MobileNavIcon({ isOpen, onClick }: { isOpen?: boolean, onClick?: (event: React.MouseEvent<HTMLDivElement>) => void }) {
+    return (
+        <div onClick={onClick} className="select-none relative w-7 h-5 mx-auto rotate-0 transition duration-500 ease-in-out pointer">
+            <span className={cn("block absolute h-[3px]  mx-auto bg-white rounded-full opacity-100  transition duration-500 ease-in-out origin-left left-[5px]", isOpen ? "rotate-45 -top-[2px]  w-full" : "rotate-0 top-0 w-2/3")}></span>
```
Comment (AndlerRL) [inline apps/web/components/layout/header/mobile-nav-icon/index.tsx:8]: @leandrogavidia -- the className `w-2/3` might be the one causing the distortion. Using relative values on pixel perfect animations may cause distortions on them, unless it is intentional for the design.

**Review by AndlerRL — APPROVED** [review]
LGTM! I add a single comment to the distortion issue with the "X" icon for the Menu Icon animation. The other observations though (design at Figma side) can also go to another PR. This one is good to go for now :+1: 

Merging!

**Pull request merged**